server/cluster.go

```diff
@@ -443,18 +443,45 @@ func (c *RaftCluster) SetStoreWeight(storeID uint64, leader, region float64) err
 }
 
 func (c *RaftCluster) checkStores() {
+	var offlineStores []*metapb.Store
+	var upStoreCount uint64
+
 	cluster := c.cachedCluster
-	for _, store := range cluster.getMetaStores() {
+	allStoresFull := true
+
+	for _, store := range cluster.GetStores() {
 		if store.GetState() != metapb.StoreState_Offline {
+			if !store.IsLowSpace(cluster.GetLowSpaceRatio()) {
+				allStoresFull = false
+			}
+			upStoreCount++
 			continue
 		}
-		if c.storeIsEmpty(store.GetId()) {
-			err := c.BuryStore(store.GetId(), false)
+		offlineStore := store.Store
+		if c.storeIsEmpty(offlineStore.GetId()) {
```

Comment:
`storeIsEmpty` need to dump all regions.  Can we improve it?

Reply:
@rleungx @nolouch Any updates about this comment?

Reply:
Got it.